app/domain/usecases/account/fetch_accounts.go

```diff
@@ -1,15 +1,21 @@
 package accountuc
 
-import "github.com/jpgsaraceni/suricate-bank/app/domain/entities/account"
+import (
+	"fmt"
+
+	"github.com/jpgsaraceni/suricate-bank/app/domain/entities/account"
+)
 
 func (uc Usecase) Fetch() ([]account.Account, error) {
 	accountList, err := uc.Repository.Fetch()
 
 	if err != nil {
-		return []account.Account{}, ErrFetchAccounts
+
+		return []account.Account{}, fmt.Errorf("repository failed to fetch accounts: %w", err)
```

Comment:
o que acha de remover a palavra "repository" da mensagem de erro?
Quem está chamando o usecase não precisa saber que foi um erro na camada de repository.
```suggestion
		return []account.Account{}, fmt.Errorf("failed to fetch accounts: %w", err)
```

Reply:
Eu tinha feito sem mesmo, mas em alguns usecases como o create_account eu tenho dois erros muito parecidos, um ao criar o account na memória e outro do repository. Aqui não tem essa diferença, só acrescentei para ficar coerente. Vou retirar aqui e alterar lá.